runtimes/runtimes/lsp/router/lspRouter.ts

```diff
@@ -42,7 +45,19 @@ export class LspRouter {
             },
         }
 
-        let responsesList = await Promise.all(this.servers.map(s => s.initialize(params, token)))
+        let responsesList = await Promise.all(
+            this.servers.map(s =>
+                s.initialize(
+                    {
+                        ...params,
+                        awsRuntimeMetadata: {
+                            customUserAgent: this.getUserAgent(params.initializationOptions?.aws),
```

Comment:
I'm curious why we ask for a lot of client specific identification, something that is useful in many places (for instance https://github.com/aws/language-servers/blob/main/server/aws-lsp-codewhisperer/src/language-server/codeWhispererServer.ts#L332) and hide it inside a specific User Agent, something Servers might want to compute themselves.

Reply:
Servers still receive raw data as part of `InitializationParams`: `params` object here is received from client and extended.

Servers still can compute their own values and choose not to use this precomputed UserAgent, we can't enforce this. The benefit of injecting `customUserAgent` is to own the standard logic, that can be shared by all Servers, avoid duplication of computing the value.